base/test.jl

```diff
@@ -848,6 +848,9 @@ function testset_beginend(args, tests)
     # action (such as reporting the results)
     quote
         ts = $(testsettype)($desc; $options...)
+        # this empty loop is here to force the block to be compiled,
+        # which is needed for backtrace scrubbing to work correctly.
+        while false; end
```

Comment:
I think this should be fine. If we change the front-end heuristics, we'll worry about this then. Probably by then backtrace scrubbing will be decoupled from compilation anyways.